src/middleware/preAuth.middleware.ts

```diff
@@ -27,6 +27,12 @@ export class PreauthMiddleware implements NestMiddleware {
           if (userExists) {
             role = userExists.role || Role.STUDENT;
           } else {
+            const newUser = new this.userModel({
+              email,
+              fid: uid,
+              role: Role.STUDENT,
+            });
+            newUser.save();
```

Comment:
you will also get photo and display name from the firebase. You can add that too.

Reply:
Sure !
I added photo url now but as per latest documentation admin.auth().verifyIdToken() no longer returns the name of the user

But the client-side can get it from localstorage as it is present there in the object and use it